### ISSUE TITLE
Allow all express-fileupload configurations

### ISSUE THREAD

**Issue by rottmann** [RobinTail/express-zod-api#1590]
Example when use limits
```ts
const serverConfig = createConfig({
  // ...
  server: {
    // ...
    upload: {
      parseNested: true,
      limits: {
        fileSize: 1024,
      },
      limitHandler: (req, res, next) => {
        return next(createHttpError(413, 'file too large'))
      }
    }
  }
}
```

throws TS2769 (No overload matches this call)

Restricted in https://github.com/RobinTail/express-zod-api/blob/master/src/config-type.ts#L78

**Comment by RobinTail** [comment]
Fair.
The `limits` should be there.
Not sure how exactly it will work with the `limitHandler`.
Need to check it.
Thank you for reporting this issue, @rottmann

**Comment by RobinTail** [comment]
`parseNested` can only be allowed if it does not conflict with JSON body parser.
I need to check it as well.

**Comment by RobinTail** [comment]
`parseNested` is always enabled by default, @rottmann 

https://github.com/RobinTail/express-zod-api/blob/d1dc43e9ed22210fac9aa5f11264011e4a4af9bf/src/server.ts#L59

**Comment by RobinTail** [comment]
@rottmann , released v17.1.0

- `limits` enabled,
- `limitError` instead of `limitHandler`.

Thank you for the contribution.

**Comment by RobinTail** [comment]
Fixed status code from `limitError` in v17.1.1
